scripts/lib/CIME/XML/machines.py

```diff
@@ -95,8 +95,34 @@ def probe_machine_name(self):
         Find a matching regular expression for hostname
         in the NODENAME_REGEX field in the file.   First match wins.
         """
-        machine = None
+
+        names_not_found = []
+
         nametomatch = socket.getfqdn()
+        machine = self._probe_machine_name_one_guess(nametomatch)
```

Comment:
I hadn't before, but I just did: From yellowstone, I tested both --machine yellowstone and --machine cheyenne, and confirmed that MACH was set correctly.